### PR TITLE
{tools}[GCC/4.9.2,foss/2015a] git and Mercurial + deps (REVEIW)

### PR DESCRIPTION
**Pull request by gjbex**
* git 1.8.5.6
* git 2.2.2
* Mercurial 3.2.4
* gettext 0.19.4
* expat 2.1.0 (GCC/4.9.2)

### CODE REVIEW

**Comment by gjbex** [comment]
Test report by @gjbex
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/2b3cce6b6e31458ba0aa for a full test report.

File: easybuild/easyconfigs/g/git/git-1.8.5.6-GCC-4.9.2.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+easyblock = 'ConfigureMake'
+
+name = 'git'
+version = '1.8.5.6'
+
+homepage = 'http://git-scm.com/'
+description = """Git is a free and open source distributed version control system designed
+to handle everything from small to very large projects with speed and efficiency."""
+toolchain = {'name': 'GCC', 'version': '4.9.2'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/git/git-1.8.5.6-GCC-4.9.2.eb]: insert newline above

File: easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+easyblock = 'ConfigureMake'
+
+name = 'git'
+version = '2.2.2'
+
+homepage = 'http://git-scm.com/'
+description = """Git is a free and open source distributed version control system designed
+to handle everything from small to very large projects with speed and efficiency."""
+toolchain = {'name': 'GCC', 'version': '4.9.2'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb]: new line again

File: easybuild/easyconfigs/m/Mercurial/Mercurial-3.2.4-foss-2014a-Python-2.7.9.eb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+easyblock = "PythonPackage"
+
+name = 'Mercurial'
+version = '3.2.4'
+
+homepage = 'http://mercurial.selenic.com/'
+description = """Mercurial is a free, distributed source control management tool. It efficiently handles projects
+of any size and offers an easy and intuitive interface.
+"""
+toolchain = {'name': 'foss', 'version': '2015a'}
```
Comment (wpoely86) [inline easybuild/easyconfigs/m/Mercurial/Mercurial-3.2.4-foss-2014a-Python-2.7.9.eb]: newline again

**Comment by wpoely86** [comment]
ok to test

**Comment by wpoely86** [comment]
with the expection of some tiny remarks, this is good to go

File: easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+name = 'git'
+version = '2.2.2'
+
+homepage = 'http://git-scm.com/'
+description = """Git is a free and open source distributed version control system designed
+to handle everything from small to very large projects with speed and efficiency."""
+
+toolchain = {'name': 'GCC', 'version': '4.9.2'}
+
+# eg. http://git-core.googlecode.com/files/git-1.8.2.tar.gz
```
Comment (boegel) [inline easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb]: drop comment, out of sync with version

File: easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+sources = ['v%s.tar.gz' % version]
```
Comment (boegel) [inline easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb]: use `%(version)s`

File: easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+source_urls = ['https://github.com/git/git/archive']
+
+preconfigopts = 'make configure && '
```
Comment (boegel) [inline easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb]: so the correct sequency is `make configure && ./configure && make && make install`?

oh man, the madness, if even git can't get this right... >_<

File: easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+]
+
+sanity_check_paths = {
+    'files': ['bin/git'],
+    'dirs': [],
+}
+
+# Work around git build system bug.  If LIBS contains -lpthread, then configure
+# will not append -lpthread to LDFLAGS, but Makefile ignores LIBS.
+configopts = "--enable-pthreads='-lpthread'"
```
Comment (boegel) [inline easybuild/easyconfigs/g/git/git-2.2.2-GCC-4.9.2.eb]: keep this closer to `preconfigopts` please

**Comment by boegel** [comment]
@gjbex: Jenkins reports:

```
AssertionError: Filename .../Mercurial-3.2.4-foss-2014a-Python-2.7.9.eb' of parsed easconfig matches expected filename 'Mercurial-3.2.4-foss-2015a-Python-2.7.9.eb'
```

typo in toolchain version in easyconfig filename

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (5 easyconfigs in this PR)
Linux SL 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f5a5258bdf4862b8a0cd for a full test report.

**Comment by boegel** [comment]
Looking good, going on, thanks @gjbex, and @wpoely86 for reviewing!

**Pull request merged**